redbot/core/core_commands.py

```diff
@@ -1940,6 +1941,357 @@ async def reload(self, ctx: commands.Context, *cogs: str):
+                com_type = discord.AppCommandType(com_type)
+                if com_type is discord.AppCommandType.message:
+                    to_add_message.append(name)
+                elif com_type is discord.AppCommandType.user:
+                    to_add_user.append(name)
+
+        # Check that we are going to enable at least one command, for user feedback
+        if not (to_add_slash or to_add_message or to_add_user):
+            await ctx.send(
+                _(
+                    "Couldn't find any disabled commands from the cog `{cog_name}`. Use `{prefix}slash list` to see all cogs with slash commands."
+                ).format(cog_name=cog_name, prefix=ctx.prefix)
+            )
+            return
+
+        SLASH_CAP = 100
+        CONTEXT_CAP = 5
+        total_slash = len(enabled_commands["slash"]) + len(to_add_slash)
+        total_message = len(enabled_commands["message"]) + len(to_add_message)
+        total_user = len(enabled_commands["user"]) + len(to_add_user)
+
+        # If enabling would exceed any limit, exit early to not enable only a subset
+        if total_slash > SLASH_CAP:
+            await ctx.send(
+                _(
+                    "Enabling all application commands from that cog would enable a total of {count} "
+                    "commands, exceeding the {cap} command limit for slash commands. "
+                    "Disable some commands first."
+                ).format(count=total_slash, cap=SLASH_CAP)
+            )
+            return
+        if total_message > CONTEXT_CAP:
+            await ctx.send(
+                _(
+                    "Enabling all application commands from that cog would enable a total of {count} "
+                    "commands, exceeding the {cap} command limit for message commands. "
+                    "Disable some commands first."
+                ).format(count=total_message, cap=CONTEXT_CAP)
+            )
+            return
+        if total_user > CONTEXT_CAP:
+            await ctx.send(
+                _(
+                    "Enabling all application commands from that cog would enable a total of {count} "
+                    "commands, exceeding the {cap} command limit for user commands. "
+                    "Disable some commands first."
+                ).format(count=total_user, cap=CONTEXT_CAP)
+            )
+            return
+
+        # Enable the cogs
+        for name in to_add_slash:
+            await self.bot.enable_app_command(name, discord.AppCommandType.chat_input)
+        for name in to_add_message:
+            await self.bot.enable_app_command(name, discord.AppCommandType.message)
+        for name in to_add_user:
+            await self.bot.enable_app_command(name, discord.AppCommandType.user)
+
+        # Update the tree with the new list of enabled cogs
+        await self.bot.tree.red_check_enabled()
+        await ctx.send(_("Enabled all commands from `{cog_name}`.").format(cog_name=cog_name))
```

Comment:
I would like to see this list how many top level commands were added and list the names of the top level commands added.